packages/wonder-blocks-form/src/components/text-field.js

```diff
@@ -78,39 +93,56 @@ export default class TextField extends React.Component<Props, State> {
         focused: false,
     };
 
-    handleOnChange: (event: SyntheticInputEvent<HTMLInputElement>) => mixed = (
-        event,
-    ) => {
-        const {validation, onValidation, onChange} = this.props;
-        const newValue = event.target.value;
+    componentDidMount() {
+        this.maybeValidate(this.props.value);
+    }
```

Comment:
Ideally we **do** want to fire `onValidation` on the initial render mainly because we would want to show any error message if the initial value for the TextField is invalid.

How do you think we could go about ensuring that the `error` state is set on the server-side, but still be able to call `onValidation` on the first render? My attempt is to replace line 99 on your suggested change with just calling my `maybeValidate()` method, but that does not seem to update the `error` state because I believe `setState` does not work inside a constructor.

Reply:
**UPDATE:**
Doing someting like this would also work...
```js
constructor(props: Props) {
    super(props);
    if (props.validation) {
        this.state.error = props.validation(props.value) || null;
        if (props.onValidation) {
            props.onValidation(this.state.error);
        }
    }
}
```
Thoughts (between all possible options)?

Reply:
Yeah, I like your proposal to set the error state in the constructor and then call the onValidation callback inside componentDidMount. I'm not sure about calling callbacks inside a constructor but that feels like it could cause problems, somehow? (Some sort of loop that is constantly re-triggering renders.) so handling it in componentDidMount feels more correct, but at least we'll get the correct error state in the text field on the initial render!